Framework/Automation.rb

```diff
@@ -0,0 +1,195 @@
+        email=tests_object.get_element_by_id("Email")
+        tests_object.send_text("test@test.net",email)
+        email=tests_object.get_element_by_id("CreatePassword")
+        tests_object.send_text("qwerty123",email)
+        form=tests_object.get_element_by_id("create_customer")
+        buttons=tests_object.get_element_by_tag("button",form)[0].click()
+
+        #logout
+        
+        tests_object.get_element_by_id("HeaderAccountToggle").click()
+        dropdown=tests_object.get_element_by_id("HeaderAccountMenu")
+        lists=tests_object.get_element_by_tag("a",dropdown)[5].click()
+        puts "done with automation of login and signup module"
+        sleep(10)
+        tests_object.navigate_to_and_maximize("https://sublimelife.in/")
+        
+    end
+
+    def search
+        puts "started with automation of search module"
+        form = tests_object.get_element_by_tag("form")[1]
+        inputs = tests_object.get_element_by_tag("input", form)[0]
+        tests_object.send_text(search_term, inputs)
+        tests_object.get_element_by_xpath(search_bar_xpath).click
+        sleep(15)
+        tests_object.close_popup(popup,"//*[@id='form-body-main']/div[1]/div/div")
+        sleep(5)
+        filter=tests_object.get_elements_by_class("snize-product-filters-list")[0]
+        items=tests_object.get_element_by_tag("li",filter)[0]
+        puts tests_object.get_element_by_xpath("//*[@id='snize_filters_block_snize_facet1']/div").click
+        puts "end with automation of add to cart module"
+        sleep(10)
+        tests_object.navigate_to_and_maximize("https://sublimelife.in/")
+    end    
+
+    def add_to_cart
+        puts "started with automation of add to cart module"
+        tests_object.scroll(2000)
+        container=tests_object.get_elements_by_class("slick-track")[2]
+        buttons=tests_object.get_element_by_tag("button",container)
+        puts  buttons[0].click
+        sleep(5)
+        c=tests_object.get_element_by_id("QuickView-martiderm-dsp-bright")
+        f=tests_object.get_element_by_tag("form",c)[0]
+        b=tests_object.get_element_by_tag("button",f);
+        b[1].click()
+        sleep(3)
+        b[2].click
+        sleep(10)
+        tests_object.get_element_by_xpath("//*[@id='ajaxCartForm']/div[3]/a").click()
+        sleep(5)
+        puts "finished with add_to_cart module"
+        sleep(10)
+        tests_object.navigate_to_and_maximize("https://sublimelife.in/")
+    end
+
+    def checkout_payment
+        puts "started testing the checkout module"
+        tests_object.scroll(2000)
+        container=tests_object.get_elements_by_class("slick-track")[2]
+        buttons=tests_object.get_element_by_tag("button",container)
+        puts  buttons[0].click
+        sleep(5)
+        c=tests_object.get_element_by_id("QuickView-martiderm-dsp-bright")
+        f=tests_object.get_element_by_tag("form",c)[0]
+        b=tests_object.get_element_by_tag("button",f);
```

Comment:
Variable names should be descriptive of the type of data it holds. Rename variables appropriately